File: pkg/model/stock_in.go
package model

import "time"

type StockIn struct {
	ID          uint
	CreatedBy   User
	Items       []StockInItem
	Status      string
	CreatedAt   time.Time
	UpdatedAt   time.Time
	FinalizedAt *time.Time
}

type StockInItem struct {
	ID            uint
	StockInID     uint
	Item          Item
	BuyPrice      float64
	TotalQuantity float64
	Packagings    []StockInPackaging
	CreatedAt     time.Time
	UpdatedAt     time.Time
}

type StockInPackaging struct {
	ID            uint
	StockInItemID uint
	ItemPackaging ItemPackaging
	Quantity      int
}

File: client/grao-a-grao/src/util/factory/stock_in.ts
import { StockInModel, StockInItemModel, StockInPackagingModel } from "@/types/stock_in";

/**
 * Creates a new, empty packaging entry for a StockInItem.
 */
export function createEmptyStockInPackaging(): StockInPackagingModel {
  return {
    id: undefined,
    item_packaging: {
      id: undefined,
      description: "",
      quantity: 0,
    },
    quantity: 1, // default to one packaging unit
  };
}

/**
 * Creates a new, empty StockInItem with a single empty packaging.
 */
export function createEmptyStockInItem(): StockInItemModel {
  return {
    id: undefined,
    item: {
      id: undefined,
      description: "",
      is_fractionable: true,
    },
    buy_price: 0,
    total_quantity: 0,
    packagings: [createEmptyStockInPackaging()],
  };
}

/**
 * Creates a new, empty StockIn with one empty item.
 */
export function createEmptyStockIn(): StockInModel {
  return {
    id: undefined,
    status: "draft",
    items: [createEmptyStockInItem()],
    created_at: undefined,
    updated_at: undefined,
  };
}
